### PR TITLE
add rke2-windows provider

### PR DESCRIPTION
**Pull request by rosskirkpat**
add rke2-windows provider to allow backend/UI to be able to discern what is/is not a windows cluster

### CODE REVIEW

**Comment by rosskirkpat** [comment]
@deniseschannon

File: providers/rke2-windows.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const RKE2_WINDOWS = "rke2.windows"
+
+
+func IsRKE2Windows(ctx context.Context, k8sClient kubernetes.Interface) (bool, error) {
+	// Look for nodes that have a Windows specific label
+	if isRKE2, err := IsRKE2(ctx, k8sClient); err != nil || isRKE2 {
```
Comment (StrongMonkey) [inline providers/rke2-windows.go:15]: IsRKE2 is going to be called twice. Can we just check if the k8s version has rke2 and it has windows nodes?

Comment (dramich) [reply providers/rke2-windows.go:15]: I told him to use this since it also has a harvester check in it. (which may or may not be important in the future)

Comment (deniseschannon) [reply providers/rke2-windows.go:15]: It will be important in the future.

Comment (dramich) [reply providers/rke2-windows.go:15]:
```suggestion
	if isRKE2, err := IsRKE2(ctx, k8sClient); err != nil || !isRKE2 {
```

File: providers/rke2-windows.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const RKE2_WINDOWS = "rke2.windows"
+
+
+func IsRKE2Windows(ctx context.Context, k8sClient kubernetes.Interface) (bool, error) {
+	// Look for nodes that have a Windows specific label
```
Comment (dramich) [inline providers/rke2-windows.go:14]: This comment should be updated to explain the rke2 check then duplicated down below

File: providers/rke2.go [hunk 3]
```diff
@@ -10,6 +11,19 @@ import (
 const RKE2 = "rke2"
 
 func IsRKE2(ctx context.Context, k8sClient kubernetes.Interface) (bool, error) {
+	// if there are windows nodes then this should not be counted as rke2.linux
+
```
Comment (dramich) [inline providers/rke2.go:15]: Extra line

File: providers/rke2-windows.go (hunk 1 above)
Comment (dramich) [reply providers/rke2-windows.go:15]: Based on the comment from Denise we should also add a harvester check in here as well

Comment (rosskirkpat) [reply providers/rke2-windows.go]: added the harvester check for rke2-windows

File: providers/rke2-windows.go (hunk 2 above)
Comment (rosskirkpat) [reply providers/rke2-windows.go]: updated w/ comments

**Review by dramich — APPROVED** [review]

**Pull request merged**